addons/mail/models/mail_thread.py

```diff
@@ -2821,11 +2821,10 @@ def _notify_by_email_get_base_mail_values(self, message, additional_values=None)
                 ('model', '=', message_sudo.model), ('res_id', '=', message_sudo.res_id),
                 ('message_type', 'in', outgoing_types),
                 ('id', '!=', message_sudo.id),
-                ('subtype_id', '!=', note_type.id),  # filters out notes, using subtype which is indexed
             ], limit=16, order='id DESC',
         )
         # filter out internal messages that are not notes, manually because of indexes
-        ancestors = ancestors.filtered(lambda m: not m.is_internal and m.subtype_id and not m.subtype_id.internal)[:3]
+        ancestors = ancestors.filtered(lambda m: not m.is_internal and m.subtype_id)[:3]
```

Comment:
Well then if you have a long list of notes, you loose real ancestors. On some records we have quite a lot of notes and few discussions so that seems a bit meh. Maybe we could increase the search limit, then sort based on subtype, and if we don't have enough, add notes ?